src/Core/Ripgrep.re

```diff
@@ -271,18 +272,33 @@ let search =
 };
 
 let findInFiles =
-    (~executablePath, ~directory, ~query, ~onUpdate, ~onComplete, ~onError) => {
+    (
+      ~executablePath,
+      ~searchExclude,
+      ~directory,
+      ~query,
+      ~onUpdate,
+      ~onComplete,
+      ~onError,
+    ) => {
+  let args =
+    searchExclude
+    |> List.map(x => "!" ++ x)
+    |> List.map(x => ["-g", x])
+    |> List.concat
+    |> List.append([
+         "--fixed-strings",
+         "--smart-case",
+         "--hidden",
+         "--json",
+         "--",
+         query,
+         directory,
+       ]);
```

Comment:
We might be able to simplify this a bit with:
```
let excludeArgs = searchExclude
   |> List.concat_map(x => ["-g", "!" ++ x]);

let args = excludeArgs @ [
	 "--fixed-strings",
         "--smart-case",
         "--hidden",
         "--json",
         "--",
         query,
         directory,
]
```

The `@` is just a short-hand for append, and the `concat_map` is a more efficient combination of `map` and `concat`: https://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html